File: rec-core/src/test/java/net/kimleo/rec/repository/selector/expr/SelectorParserTest.java
package net.kimleo.rec.repository.selector.expr;

import net.kimleo.rec.repository.selector.AliasSelector;
import net.kimleo.rec.repository.selector.FieldSelector;
import net.kimleo.rec.repository.selector.Selector;
import org.junit.Test;

import java.util.List;
import java.util.Objects;

import static net.kimleo.rec.repository.selector.expr.SelectorParser.*;
import static org.junit.Assert.*;

public class SelectorParserTest {

    @Test
    public void testParser() {
        List<SelectorToken> lex = lex("cust, prod.hello world, i2b[name, pass], intj");
        List<Selector> parse = parse(lex);

        assertTrue(parse.size() == 4);
        assertTrue(((FieldSelector) parse.get(0)).properties.size() == 1);
        assertTrue(Objects.equals(((FieldSelector) parse.get(1)).properties.get(0), "hello world"));
        assertTrue(((FieldSelector) parse.get(2)).properties.size() == 2);
    }

    @Test
    public void testAlias() {
        List<SelectorToken> lex = lex("cust[name, age] as person");
        List<Selector> parse = parse(lex);

        assertTrue(parse.size() == 1);

        assertEquals(((AliasSelector) parse.get(0)).alias, "person");
    }

}

File: rec-core/src/test/kotlin/net/kimleo/rec/repository/selector/expr/SelectorParserTest.kt
package net.kimleo.rec.repository.selector.expr

import net.kimleo.rec.repository.selector.AliasSelector
import net.kimleo.rec.repository.selector.FieldSelector
import net.kimleo.rec.repository.selector.expr.SelectorParser.lex
import net.kimleo.rec.repository.selector.expr.SelectorParser.parse
import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test

class SelectorParserTest {
    @Test
    fun testParser() {
        val lex = lex("cust, prod.hello world, i2b[name, pass], intj")
        val parse = parse(lex)

        assertTrue(parse.size == 4)
        assertTrue((parse[1] as FieldSelector).properties.size == 1)
        assertTrue((parse[1] as FieldSelector).properties[0] == "hello world")
        assertTrue((parse[2] as FieldSelector).properties.size == 2)
    }

    @Test
    fun testAlias() {
        val lex = lex("cust[name, age] as person")
        val parse = parse(lex)

        assertTrue(parse.size == 1)

        assertEquals((parse[0] as AliasSelector).alias, "person")
    }
}